File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.cs
using System;

public class Solution {
    public string solution(string[] cards1, string[] cards2, string[] goal) {
       string answer = "Yes";
        int index1 = 0;
        int index2 = 0;
        
        for(int i = 0; i < goal.Length; i++)
        {
            if(index1 < cards1.Length && goal[i] == cards1[index1])
            {
                index1++;
                continue;
            }
            else if(index2 < cards2.Length && goal[i] == cards2[index2])
            {
                index2++;
                continue;
            }
            else
            {
                return "No";
            }         
        }
        
        return answer;
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    string answer = "Yes";
    int first = 0;
    int second = 0;
    for(int i  = 0; i < goal.size(); i++)
    {
        if(goal[i] == cards1[first])
        {
            first++;
        }
        else if(goal[i] == cards2[second])
        {
            second++;
        }else
        {
            answer = "No";
            break;
        }
        
    }
    
    return answer;
}